File: Shadoks.cs
using System;
using System.Text;
using System.Collections.Generic;

class Shadoks{
    static public readonly String[] Alphabet = {"Ga","Bu","Zo","Meu"};
    static public string Traduire(int num){
        List<int> indexs = new List<int>();
        StringBuilder text = new StringBuilder();
        while(true){
            text.Insert(0,Alphabet[num%4]);
            if(num>3){
                num = num/4;
            }else{
                break;
            }
        }
        return text.ToString();
    }

    static public int Traduire(string text){
        double compteur = 0;
        double num = 0;
        while(true){
            bool fini = true;
            foreach(string lettre in Alphabet){
                if(text.EndsWith(lettre)){
                    num += Array.IndexOf(Alphabet,lettre)*(Math.Pow(4d,compteur++));
                    text = text.Substring(0,text.Length-lettre.Length);
                    fini = false;
                }
            }
            if(fini)break;
        }
        return (int)num;
    }
}

File: Shadoks.java
import java.util.Arrays;

public class Shadoks {
    static final String[] Alphabet = {"Ga","Bu","Zo","Meu"};
    
    static public String Traduire(int num){
        StringBuilder text = new StringBuilder();
        while(true){
            text.insert(0, Alphabet[num%4]);
            if(num>3){
                num = num/4;
            }else{
                break;
            }
        }
        return text.toString();
    }
    
    static public int Traduire(String text){
        int compteur = 0, num = 0;
        while(true){
            boolean fini = true;
            for(String lettre : Alphabet){
                if(text.endsWith(lettre)){
                    num += Arrays.asList(Alphabet).indexOf(lettre)*(Math.pow(4d,compteur++));
                    text = text.substring(0,text.length()-lettre.length());
                    fini = false;
                }
            }
            if(fini)break;
        }
        return num;
    }
    
}